client/src/periodUtils.js

```diff
@@ -42,16 +42,28 @@ export const PERIOD_FACTORIES = {
       .subtract(2 * offset, "weeks")
       .endOf("week")
   }),
-  [RECURRENCE_TYPE.SEMIMONTHLY]: (date, offset) => ({
-    start:
-      date.date() < 15
-        ? date.clone().subtract(offset, "months").startOf("month")
-        : date.clone().subtract(offset, "months").startOf("month").date(15),
-    end:
-      date.date() < 15
-        ? date.clone().subtract(offset, "months").startOf("month").date(14)
-        : date.clone().subtract(offset, "months").endOf("month")
-  }),
+  [RECURRENCE_TYPE.SEMIMONTHLY]: (date, offset) => {
+    // Don't subtract 15 days to cover edge cases: Febuary being 28-29 days and some months 31 days
+    const baseDate =
+      offset % 2 === 0
+        ? date.clone().subtract(offset / 2, "months")
+        : date
+          .clone()
+          .subtract(Math.ceil(offset / 2), "months")
+          .add(15, "days")
+
+    const isFirstHalfOfTheMonth = baseDate.date() < 16
+    const baseMonthStart = baseDate.clone().startOf("month")
+
+    return {
+      start: isFirstHalfOfTheMonth
+        ? baseMonthStart
+        : baseMonthStart.clone().add(15, "days"), // start of day 16
```

Comment:
This seems to imply we start on the 16th, when we want to start on the 15th

Reply:
Are you sure you want to start from 15th? If you do so it will look like this:
`1st-14th Jan => 15th-30th Jan` (not really symmetric, 14 days in the first half, 16 days in the second)

Reply:
I know it is arbitrary, but people tend to start on the 15th when it is mid-month

Reply:
Sure, i will fix the other issue as well